### PR TITLE
Fix spacing for inline widgets

### PR DESCRIPTION
**Pull request by DCsunset**
I have added a spacing setting for each widget that can control the spacing.
The `Inline` setting will remove the leading space, which is very useful for sub-widget in some blocks.
The `Hidden` setting will remove both the leading space and trailing space, which is useful for hidden blocks.

To demonstrate the usage, I have also fixed the spacing for `nvidia_gpu` block.

This PR should resolve the issue #800.

### CODE REVIEW

**Comment by ammgws** [comment]
Can you paste the before and after nvidia-gpu block output here (i.e. the raw output from i3status-rs when run from a commandline)?

File: src/widget.rs [hunk 1]
```diff
@@ -7,6 +7,15 @@ use serde_json::value::Value;
 
 use crate::themes::Theme;
 
+#[derive(Debug, Copy, Clone, Deserialize)]
+pub enum Spacing {
+    Normal,
+    /// Hide the leading space when the widget is inline
+    Inline,
+    /// Hide all spaces when the widget is hidden
```
Comment (ammgws) [inline src/widget.rs:15]: Just to make it clearer: "Hide both leading and trailing spaces when widget is hidden"

File: src/widget.rs [hunk 2]
```diff
@@ -7,6 +7,15 @@ use serde_json::value::Value;
 
 use crate::themes::Theme;
 
+#[derive(Debug, Copy, Clone, Deserialize)]
+pub enum Spacing {
+    Normal,
```
Comment (ammgws) [inline src/widget.rs:12]: Just to make it clearer: "Adds a leading and trailing space around the widget contents"

**Review by ammgws — CHANGES_REQUESTED** [review]

**Comment by DCsunset** [comment]
Before:

```json
{"background":"#000000FF","color":"#CCCCCCFF","full_text":"  GeForce MX150 ","markup":"pango","name":"f6a82f89f0f241dd838918b61a6deb6c","separator":false,"separator_block_width":0},
{"background":"#000000FF","color":"#CCCCCCFF","full_text":" 0 % ","separator":false,"separator_block_width":0},
{"background":"#000000FF","color":"#CCCCCCFF","full_text":" 52°C ","separator":false,"separator_block_width":0}
```

After:

```json
{"background":"#000000FF","color":"#CCCCCCFF","full_text":"  GeForce MX150 ","markup":"pango","name":"547e37d81bae45cd9e96ecfc4713ac12","separator":false,"separator_block_width":0},
{"background":"#000000FF","color":"#CCCCCCFF","full_text":"0 % ","separator":false,"separator_block_width":0},
{"background":"#000000FF","color":"#CCCCCCFF","full_text":"41°C ","separator":false,"separator_block_width":0}
```

Leading spaces of both the temperature and utilization are removed.
Besides, when the name of the GPU is hidden, its trailing spaces will also be removed.

**Comment by DCsunset** [comment]
I have also fixed the comment and squash it into the first commit.

**Comment by ammgws** [comment]
Can you please apply this to the other blocks as well? Might as well fix them all at once.

**Review by ammgws — APPROVED** [review]

**Review by ammgws — CHANGES_REQUESTED** [review]
Apply to other blocks as well. Might as well fix them all at once.

**Comment by DCsunset** [comment]
I have fixed some blocks that I found with inconsistent spacing. Are there other blocks that need fixing as well?

**Comment by ammgws** [comment]
IIRC one would be the music block (the buttons at least).

**Review by ammgws — APPROVED** [review]

**Comment by ammgws** [comment]
We can handle those in another PR

**Pull request merged**